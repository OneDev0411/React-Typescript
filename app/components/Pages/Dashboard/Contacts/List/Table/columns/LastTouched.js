import React from 'react'
import timeago from 'timeago.js'
import { Popover, OverlayTrigger } from 'react-bootstrap'

export function LastTouchedCell(props) {
  const { contact } = props

  if (!contact.last_touch) {
    return null
  }

  const formatedLastTouch = timeago().format(contact.last_touch * 1000)

  if (!contact.next_touch) {
    return formatedLastTouch
  }

  return (
    <OverlayTrigger
      trigger="hover"
      placement="top"
      overlay={
        <Popover
          id={`last-touch-popover_${contact.id}`}
          style={{
            color: '#fff',
            border: 'none',
            backgroundColor: '#303e4d'
          }}
        >
          <span>Last touch was </span>
          <b>{formatedLastTouch}, </b>
          <span>you wanted to be in touch every 7 days.</span>
        </Popover>
      }
    >
      <span>{formatedLastTouch}</span>
    </OverlayTrigger>
  )
}
