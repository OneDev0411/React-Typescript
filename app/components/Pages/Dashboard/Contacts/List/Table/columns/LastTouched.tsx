import { makeStyles, Theme, Tooltip } from '@material-ui/core'
import timeago from 'timeago.js'

interface Props {
  contact: IContact
}

const useStyles = makeStyles(
  (theme: Theme) => ({
    noTouch: {
      color: theme.palette.grey[700],
      ...theme.typography.caption
    },
    lastTouchValue: theme.typography.body2
  }),
  { name: 'LastTouchedCell' }
)

export default function LastTouched({ contact }: Props) {
  const classes = useStyles()
  const { last_touch: lastTouch, next_touch: nextTouch } = contact

  if (!lastTouch) {
    return <span className={classes.noTouch}>No Touches</span>
  }

  const formattedLastTouch = timeago().format(lastTouch * 1000)

  return (
    <div>
      {nextTouch ? (
        <Tooltip
          title={
            <span>
              You wanted to be in touch
              <br />
              every {Math.round((nextTouch - lastTouch) / 86400)} days.
            </span>
          }
        >
          <span className={classes.lastTouchValue}>{formattedLastTouch}</span>
        </Tooltip>
      ) : (
        <span className={classes.lastTouchValue}>{formattedLastTouch}</span>
      )}
    </div>
  )
}
