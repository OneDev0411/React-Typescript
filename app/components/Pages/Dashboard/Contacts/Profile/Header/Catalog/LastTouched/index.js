import React from 'react'
import timeago from 'timeago.js'

export function LastTouched(props) {
  const { last_touch, next_touch } = props.contact

  return (
    <div>
      {last_touch ? (
        <div>
          <span>
            Last Touch was <b>{timeago().format(last_touch * 1000)}</b>
          </span>
          {next_touch ? (
            <span>
              , you wanted to be in touch every{' '}
              <b>{(next_touch - last_touch) / 86400}</b> days.
            </span>
          ) : (
            '.'
          )}
        </div>
      ) : (
        <div>You have not added any touches for this contact.</div>
      )}
    </div>
  )
}
