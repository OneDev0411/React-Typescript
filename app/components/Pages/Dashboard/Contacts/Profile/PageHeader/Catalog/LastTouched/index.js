import React from 'react'
import timeago from 'timeago.js'

export function LastTouched(props) {
  const { last_touch } = props.contact

  return (
    <div>
      {last_touch ? (
        <div>
          Last Touch was <b>{timeago().format(last_touch * 1000)}</b>.
        </div>
      ) : (
        <div>You have not added any touches for this contact.</div>
      )}
    </div>
  )
}
