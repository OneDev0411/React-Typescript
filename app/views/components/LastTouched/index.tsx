import React, { ReactNode } from 'react'

import { Typography, Box, Theme } from '@material-ui/core'
import { makeStyles, createStyles } from '@material-ui/styles'
import timeago from 'timeago.js'

import { frequencyToString } from '@app/components/Pages/Dashboard/Contacts/components/ManageRelationship/helper'

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    wrapper: {
      color: theme.palette.grey[900],
      ...theme.typography.body2
    },
    label: {
      color: theme.palette.grey[900]
    }
  })
)

interface Props {
  contact: IContact
  children?: ReactNode
}

export function LastTouched({ contact, children }: Props) {
  const classes = useStyles()
  const { last_touch: lastTouch, touch_freq: touchFreq } = contact

  if (!lastTouch) {
    return (
      <Typography
        variant="body2"
        className={classes.label}
        data-tour-id="last-touch"
      >
        You have not added any touches for this contact.
      </Typography>
    )
  }

  return (
    <Box className={classes.wrapper} data-tour-id="last-touch">
      <span>
        Last Touch was <b>{timeago().format(lastTouch * 1000)}</b>
        {!touchFreq && '.'}
      </span>
      {touchFreq && (
        <span>
          , you wanted to be in touch{' '}
          <b>{frequencyToString(touchFreq || null).toLocaleLowerCase()}</b>
        </span>
      )}
      {children}
    </Box>
  )
}
