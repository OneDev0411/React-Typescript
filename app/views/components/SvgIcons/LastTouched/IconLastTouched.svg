<svg xmlns="http://www.w3.org/2000/svg" width="24" height="23" viewBox="0 0 24 23">
    <g fill="none">
        <path fill="none" d="M0-1h24v24H0z"/>
        <g class="c-icon-last-touched">
            <g transform="translate(6.492 5.155)">
                <path d="M5.469.08A5.243 5.243 0 0 0 .315 4.341c-.06.357.157.694.511.754a.634.634 0 0 0 .748-.516c.354-1.864 1.987-3.232 3.875-3.232 2.184 0 3.954 1.785 3.954 3.986 0 2.2-1.77 3.985-3.954 3.985-.452 0-.885-.08-1.318-.218a.65.65 0 0 0-.826.396.657.657 0 0 0 .393.833c.551.198 1.141.298 1.751.298 2.892 0 5.253-2.38 5.253-5.294A5.234 5.234 0 0 0 5.469.079z"/>
                <path d="M6.197 5.85l1.77-1.389-.137-.337-2.223.357c-.04 0-.099-.02-.138-.02a.885.885 0 0 0-.885.892c0 .496.393.893.885.893.295 0 .57-.159.728-.397z"/>
                <ellipse cx="2.164" cy="8.585" rx="1" ry="1"/>
                <ellipse cx="1.015" cy="6.503" rx="1" ry="1" transform="rotate(-59.935 1.015 6.503)"/>
            </g>
            <path d="M23.705 18.638c-.216-.535-.433-1.17-.55-1.467l-.06-.179c-.806-2.161-1.082-4.5-1.24-6.503v-.06c-.176-2.16-.353-4.401-1.77-6.325A9.684 9.684 0 0 0 12.275.12h-.55c-3.07 0-6 1.487-7.83 3.985-1.416 1.943-1.593 4.184-1.77 6.385-.158 1.983-.433 4.322-1.24 6.503l-.059.199c-.118.317-.315.931-.53 1.467a1.72 1.72 0 0 0 .156 1.546c.296.456.807.734 1.358.734h7.082A3.448 3.448 0 0 0 12 22.9a3.49 3.49 0 0 0 3.128-1.963h7.082c.531 0 1.042-.278 1.357-.734.295-.476.354-1.05.138-1.566zm-21.777.535c.216-.535.413-1.11.531-1.427l.059-.159c.885-2.38 1.18-4.858 1.338-7 .157-2.021.314-3.925 1.436-5.471 1.495-2.063 3.915-3.272 6.452-3.272h.551c2.538 0 4.938 1.23 6.453 3.291 1.12 1.527 1.278 3.43 1.436 5.433v.06c.177 2.121.472 4.6 1.337 6.979l.06.159c.098.297.314.872.53 1.427H1.928v-.02z"/>
        </g>
    </g>
</svg>
