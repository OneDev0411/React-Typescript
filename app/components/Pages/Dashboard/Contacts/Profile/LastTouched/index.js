import React from 'react'
import timeago from 'timeago.js'
import Flex from 'styled-flex-component'

import { Section } from '../components/Section'
import Icon from '../../../../../../views/components/SvgIcons/LastTouched/IconLastTouched.js'

export function LastTouched(props) {
  const { last_touch } = props.contact

  return (
    <Section
      title="Last Touched"
      bodyStyle={{
        padding: '1em',
        fontSize: '1.6rem',
        color: last_touch ? '#17283a' : '#8da2b5'
      }}
    >
      <Flex alignCenter>
        <Icon
          style={{
            marginRight: '1em',
            fill: last_touch ? '#17283a' : '#8da2b5'
          }}
        />
        {last_touch ? (
          <div>
            Last Touch was <b>{timeago().format(last_touch * 1000)}</b>.
          </div>
        ) : (
          <div>You have not added any touches for this contact.</div>
        )}
      </Flex>
    </Section>
  )
}
