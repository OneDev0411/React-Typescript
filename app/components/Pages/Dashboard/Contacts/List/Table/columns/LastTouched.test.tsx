import { render } from 'enzyme'
import toJson from 'enzyme-to-json'

import fullContact from 'fixtures/contacts/full-contact.json'
import mockDate, { RealDate } from 'utils/test-utils/mock-date'

// TODO: full-contact.json data is old and needs update
const contact: any = {
  ...fullContact,
  user: fullContact.user as IUser,
  users: fullContact.users as IUser[]
}

import LastTouched from './LastTouched'

describe('Contacts list last touched component', () => {
  beforeEach(() => {
    mockDate('2019-05-26T12:04:04Z')
  })

  afterEach(() => {
    global.Date = RealDate
  })

  it('renders for contacts with last touch', () => {
    const wrapper = render(<LastTouched contact={contact} />)

    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('renders for contacts without last touch', () => {
    contact.last_touch = null

    const wrapper = render(<LastTouched contact={contact} />)

    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
