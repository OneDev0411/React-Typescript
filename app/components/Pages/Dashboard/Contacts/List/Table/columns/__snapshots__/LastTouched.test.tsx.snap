// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contacts list last touched component renders for contacts with last touch 1`] = `
<div>
  <span
    class="LastTouchedCell-lastTouchValue-2"
  >
    just now
  </span>
</div>
`;

exports[`Contacts list last touched component renders for contacts without last touch 1`] = `
<span
  class="LastTouchedCell-noTouch-1"
>
  No Touches
</span>
`;
